src/core/sock/sockinfo_tcp.cpp

```diff
@@ -1850,14 +1840,16 @@ void sockinfo_tcp::abort_connection()
     tcp_abort(&(m_pcb));
 }
 
-int sockinfo_tcp::handle_child_FIN(sockinfo_tcp *child_conn)
+void sockinfo_tcp::handle_incoming_handshake_failure(sockinfo_tcp *child_conn)
 {
-    lock_tcp_con();
+    child_conn->unlock_tcp_con();
 
```

Comment:
why don't we leave the child_conn locked through the whole function?

Reply:
Reminder: Remove accpted_conn loop.

Reply:
Removed unneeded m_accepted_conn loop after setting m_parent=nullptr before putting the conenction into m_accepted_conn

Reply:
We do need to unlock child because we lock listen socket. It will work without, but it is not a good practice to have two locks acquired by the same thread.